<a href="https://colab.research.google.com/github/LRisaro/TP_DATOS_2021/blob/main/TP_DATOS.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [2]:
import pandas as pd
import numpy as np
from pandas_profiling import ProfileReport
from matplotlib import pyplot as plt
import seaborn as sbn
from sklearn import tree
from sklearn.metrics import accuracy_score

%matplotlib inline

plt.style.use('default') 

sbn.set(style="whitegrid") 

In [7]:
def GetData():
  return pd.read_csv("https://docs.google.com/spreadsheets/d/1-DWTP8uwVS-dZY402-dm0F9ICw_6PNqDGLmH0u8Eqa0/export?format=csv&gid=2054774482")

**Inspección inicial de datos**

In [11]:
df = GetData()
df

anios_estudiados  ...                    trabajo
0                    17  ...            entretenimiento
1                    17  ...          directivo_gerente
2                    13  ...                  limpiador
3                    11  ...                  limpiador
4                    17  ...  profesional_especializado
...                 ...  ...                        ...
32556                16  ...            soporte_tecnico
32557                13  ...                  inspector
32558                13  ...            entretenimiento
32559                13  ...            entretenimiento
32560                13  ...          directivo_gerente

[32561 rows x 13 columns]

In [12]:
# Inspeccionamos las dimensiones iniciales del dataframe
df.shape

(32561, 13)

In [14]:
# Verifico que "tiene_alto_valor_adquisitivo" no tenga valores nulos
df['tiene_alto_valor_adquisitivo'].isnull().sum()

0

In [19]:
# Verificamos si hay campos con valores nulos
df.isnull().sum()

anios_estudiados                                 0
barrio                                         583
categoria_de_trabajo                          1836
edad                                             0
educacion_alcanzada                              0
estado_marital                                   0
ganancia_perdida_declarada_bolsa_argentina       0
genero                                           0
horas_trabajo_registradas                        0
religion                                         0
rol_familiar_registrado                          0
tiene_alto_valor_adquisitivo                     0
trabajo                                       1843
dtype: int64

**Análisis de columnas**

**Categoria de trabajo**

**Preguntas:**

* ¿La columna aporta información relevante?
* ¿El dato esta completo en todas las filas?
* ¿Los datos tienen mucha varianza?
* ¿Aporta valor para realizar una predicción?

**Análisis**


In [24]:
df["categoria_de_trabajo"].value_counts()

relacion_de_dependencia    22696
monotibutista               2541
empleado_municipal          2093
empleado_provincial         1298
responsable_inscripto       1116
empleadao_estatal            960
trabajo_voluntariado          14
sin_trabajo                    7
Name: categoria_de_trabajo, dtype: int64

In [26]:
# Tiene valores nulos?
df["categoria_de_trabajo"].isnull().sum()

1836

**Conclusiones**

Podemos observar que faltan 1836 de los 32561 valores (5,6 %).

**Relación entre columnas**

**Conclusión final**

**Feature Engineering**

In [27]:
def FeatureEngineering(df):
    
    df_clean = df.copy()
            
    return df_clean

In [28]:
def Predict(df):
    pred = []
    values = df.values

    # for val in values:
    #     if val[1] == 'hombre':
    #         pred.append(0)
    #     else:
    #         if (val[0] == '4d' and val[4] == 'Palermo'):
    #             pred.append(0)
    #         else:
    #             pred.append(1)
     
    return pred

In [29]:
def baseline(df):
    """    
    Input: dataframe
    Output: lista con las predicciones    
    """    
    # Aplicamos feature engineering a los datos de entrada
    entrada = FeatureEngineering(df)
    pred = Predict(entrada)
    
    return pred

In [ ]:
# Generamos dataframe para probar
df_datos_test = GetData()
df_datos_valor_adquisitivo = df_datos_test["tiene_alto_valor_adquisitivo"]

predicciones = baseline(df_datos_test)
accuracy_score(df_datos_valor_adquisitivo, predicciones)